home/web/client/components/TransactionConfirmation/TransactionConfirmation.js

```diff
@@ -97,101 +102,102 @@ export default class TransactionConfirmation extends Web3Component {
     return <Account address={ to } />;
   }
 
-  renderPassword (from) {
-    const { password, error, sending } = this.state;
+  renderPassword () {
+    const { password, error } = this.state;
+    const { isSending } = this.props;
+    const { from } = this.props.transaction.params[0];
 
-    if (sending) {
+    if (isSending) {
       return (
         <div className={ styles.progress }>
           <LinearProgress mode='indeterminate' />
         </div>
       );
     }
 
-    const errorMsg = password ? (error ? 'Invalid password or transaction.' : null) : 'Type your password.';
+    const errorMsg = error.length ? error : null;
 
     return (
       <TextField
         fullWidth
         hintText={ `Password for ${from}` }
         errorText={ errorMsg }
-        floatingLabelText='Unlock the account'
+        floatingLabelText='Type your password'
```

Comment:
I think `Type your password` is even more inaccurate - it will be displayed only after user already typed the password (cause it's floating text label) so it doesn't make sense.
It could be `Password to unlock your account`, I don't know what will work best.